nvc/nvc.go

```diff
@@ -33,69 +33,97 @@ const (
 
 var ErrNoMagicFound error = errors.New("nvc magic bytes not found")
 
-// TocEntry is a file entry in the table of contents.
-type TocEntry struct {
-	Hash      Hash       // 64-bit FNV-1a hash of the file's path on disk
-	Offset    uint32     // File's offset (in bytes) from the start of the nvc file
-	RawLength uint32     // Length (in bytes) of the file after it has been extracted
-	Length    uint32     // Length (in bytes) of file as it is stored in the nvc file
-	Flags     EntryFlags // Indicates whether file is compressed or encrypted (TODO: determine if this is a bitmask)
+type Archive struct {
+	Entries    map[Hash]TocEntry // Map of hashes to table of contents entries
+	EntryOrder []Hash            // List of entry hashes in the order that they are stored in the archive
+
+	r io.ReadSeeker
 }
 
-// ReadToc parses r as an NVC archive and returns its table of contents.
-func ReadToc(r io.Reader) ([]TocEntry, error) {
+// Parse reads r and attempts to interpret is as an NVC archive.
+// This function takes ownership of r; it should not be used by the caller after Parse has been called.
+func Parse(r io.ReadSeeker) (Archive, error) {
```

Comment:
Let's document that a garbage struct is returned on err.  The convention of returning an instance of a data structure on error is a little confusing coming from other languages that have a different concept of no data.  https://go.dev/ref/spec#The_zero_value

Reply:
Added in 669a0a7a7d4ae82a3045ff5dc5855f9fab211dc6